### PR TITLE
Add within 3d box detection.

### PR DESCRIPTION
**Pull request by tanzhenyu**
# What does this PR do?
This PR creates functionality to detect whether 3d point clouds are within 3d bounding boxes

### CODE REVIEW

**Comment by lengzq** [comment]
+1

**Comment by tanzhenyu** [comment]
Also KerasCV authors would like to acknowledge the helpful contribution from [lingvo](https://github.com/tensorflow/lingvo), which provides most of the existing implementations, and borrowed here under their license.

File: keras_cv/ops/point_cloud.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+import tensorflow as tf
+
+
+def get_rank(tensor):
+    return tensor.shape.ndims or tf.rank(tensor)
+
+
+def _center_xyzWHD_to_corner_xyz(boxes):
+    # relative corners w.r.t to origin point
```
Comment (lengzq) [inline keras_cv/ops/point_cloud.py:23]: Please add docstrings with Args and Returns:

e.g.

      boxes: [..., num_boxes, 7] float32 Tensor for 3d boxes in [x, y, z, dx,
        dy, dz, phi].

**Review by lengzq — APPROVED** [review]

**Pull request merged**